exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/SchemaUtils.kt

```diff
@@ -624,6 +666,72 @@ object SchemaUtils {
         return toCreate.toList()
     }
 
+    /**
+     * @param scriptName The name to be used for the generated migration script.
+     * @param scriptDirectory The directory in which to create the migration script.
+     * @param withLogs By default, a description for each intermediate step, as well as its execution time, is logged at
+     * the INFO level. This can be disabled by setting [withLogs] to `false`.
+     *
+     * @return The generated migration script.
+     *
+     * This function simply generates the migration script without applying the migration. The purpose of it is to show
+     * the user what the migration script will look like before applying the migration.
+     */
+    @ExperimentalDatabaseMigrationApi
+    fun generateMigrationScript(vararg tables: Table, scriptDirectory: String, scriptName: String, withLogs: Boolean = true): File {
+        val allStatements = statementsRequiredForDatabaseMigration(*tables, withLogs = withLogs)
+
+        val migrationScript = File("$scriptDirectory/$scriptName.sql")
+        val migrationScriptExists = if (migrationScript.exists()) true else migrationScript.createNewFile()
+        if (migrationScriptExists) {
```

Comment:
We can invert the condition to reduce nesting and make it a bit readable:
```if (!migrationScriptExists) throw NoSuchFileException(migrationScript, reason = "Failed to create/find migration script")```

Reply:
Won't `migrationScriptExists` always evaluate to `true`? Because `createNewFile()` only returns `false` if the named script already exists and that branch would only be reached if the file didn't already exist (so a new one is made)?

`createNewFile()` also checks for existence of the file so maybe the `if` blocks could be combined in a `try` block that throws the exception above on any IO error?

Maybe there could also be a test for this case when a script already exists at the defined path.

Reply:
Good point. There's no need to call `migrationScript.exists()` at all. We can just call `migrationScript.createNewFile()`. There's also no need for us to throw an exception because if we don't wrap `migrationScript.createNewFile()` in a try-catch block, whatever exception is thrown there will be shown.
I will add a test for the case when a script already exists.